### ISSUE TITLE
Nest should only introduce indentation after a line break.

### ISSUE THREAD

**Issue by juliapath** [wenkokke/py-doc-printer#1]
I.e. `Text("label:") // Nest(2, Text("a") / Line / Text("b"))` should produce
```
label: a
  b
```
not
```
label:   a
  b
```

**Comment by wenkokke** [comment]
Oh, right, I've only been using `Nest` directly after a line break, so I didn't notice.

**Comment by wenkokke** [comment]
I suppose nest should introduce enough indentation at the start to make sure it starts at the necessary column, i.e., `Text("label:") // Nest(10, Text("a") / Line / Text("b"))` should produce:
```
label:    a
          b
```

...not...

```
label: a
          b
```

**Comment by juliapath** [comment]
That's how `nest` of `prettyprinter` and the others from Haskell behaves. There is also `align` which behaves similar to your first example, but is more costly for reasons unknown to me. It's also usually not what I want as it introduces deeper indentation. My preference is to use `nest` with `group` to either get everything on one line or start on a new one with a fixed indentation.

**Comment by juliapath** [comment]
Also thanks for writing this. I'm new to Python from Haskell and the library situation is killing me. This at least gives me some semblance of feeling at home.

**Comment by wenkokke** [comment]
> That's how `nest` of `prettyprinter` and the others from Haskell behaves. There is also `align` which behaves similar to your first example, but is more costly for reasons unknown to me. It's also usually not what I want as it introduces deeper indentation. My preference is to use `nest` with `group` to either get everything on one line or start on a new one with a fixed indentation.

So you're saying `nest` should behave like what you suggested, `align` could behave like what I suggested?

**Comment by wenkokke** [comment]
Should be fixed in 91f701c78c200ee9184e94a86f520838b9d88a48, version 0.6.0.

**Comment by juliapath** [comment]
> So you're saying `nest` should behave like what you suggested, `align` could behave like what I suggested?

Not exactly. `align` is like a `nest` along the current column. So that `"label: " $ align "a\nb"` yields the following:

```
label: a
       b
```

There are a few other alignment functions which are just combinations `nest` and `align`.
